### DISCUSSION TITLE
第二版のレビュアーを募集

### DISCUSSION THREAD

**Discussion by azu** [asciidwango/js-primer#1546] [General]
JavaScript Primerの第二版の出版準備を進めています。

初版からの主な変更点をまとめると、次のようになります。

- ECMAScriptの新しいバージョンであるES2020、ES2021、ES2022に対応した
- 新しいECMAScriptの機能によって、使う必要がなくなった機能は非推奨へと変更した
- 文字では想像しにくいビット演算、非同期処理などに図を追加した
- PromiseとAsync Functionを非同期の処理の中心として書き直した
- 一方で、エラーファーストコールバックは非同期処理としてはメインではなくなった
- `Array#includes`という表記は、Private Classs Fields(`#field`)と記号が被るため廃止した
- Node.jsでもECMAScript Modulesを使うようになり、CommonJSはメインではなくなった
- Node.jsが12から18までアップデートし、npmは6から8までアップデート、各種ライブラリも最新にアップデート
- 読者からのフィードバックを受けて、全体をより分かりやすく読みやすくなるように書き直した

リリースノート的には、次のものが第二版には含まれています。

- [v2.0.0: ECMAScript 2020対応](https://github.com/asciidwango/js-primer/releases/tag/v2.0.0)
- [v3.0.0: ECMAScript 2021対応](https://github.com/asciidwango/js-primer/releases/tag/v3.0.0)
- [v4.0.0: ECMAScript 2022対応](https://github.com/asciidwango/js-primer/releases/tag/v4.0.0)

かなり全体的に書き直しているため、jsprimerの第二版を~~レビューしてくれる方を募集しています。~~ <ins>レビューをしてくれる人が集まったので、一旦募集はここで一区切りにします</ins>

## 期間

- 2023年2月中いっぱい

## レビューする内容

- [ウェブ版](https://jsprimer.net/)を見て、気になったことをコメント/Issue/Pull Requestしてもらう
- レビュー用のDiscussion: https://github.com/asciidwango/js-primer/discussions/1547

## レビューアーの方へ

- ~~レビューをしてくれる方は、このDiscussionにコメント または [info@jsprimer.net](mailto:info@jsprimer.net) にメール または [@azu_re](https://twitter.com/azu_re) にMentionしてください。~~  <ins>レビューをしてくれる人が集まったので、一旦募集はここで一区切りにします</ins>
- PDFについては <https://github.com/orgs/js-primer/teams/reviewer/discussions/1> を参照してください
- レビューアーの方は[謝辞](https://jsprimer.net/intro/#thanks)に掲載させていただきます
- 出版が完了しましたら、書籍を献本させていただきます(希望者のみ)

**Comment by KeisukeKudo** [comment]
はじめまして｡
レビューアーを募集しているということで､ぜひ参加させてください

**Reply by azu** [reply]
ありがとうございます！

PDF版はまだ組版中のため途中の章までしかないですが、PDFの方でレビューしたいですかね?(内容自体は同じです)

**Reply by KeisukeKudo** [reply]
Web 版で大丈夫です!
誤字脱字や分かりづらい表現､技術的な誤りがあったら適宜 PR 出す､というような流れでいいでしょうか?

**Reply by azu** [reply]
はい！
やりやすい方法で問題ないです。
Issueを作るだけでも問題ありません。

レビュー用のDiscussionも作成したので、Issueにするべきかわからないようなものだったりしたら、こちらに書いてもらっても大丈夫です。

- https://github.com/asciidwango/js-primer/discussions/1547

**Reply by KeisukeKudo** [reply]
ありがとうございます!
書籍のレビューは初めてなので至らない点も多々あるかと思いますが､よろしくお願いします｡

**Comment by nikukyugamer** [comment]
第一版 はつねに手の届くところに置いていつでも見られるようにしています。

レビュアーとして参加させていただきたいと思います。
よろしくお願いします。

**Reply by azu** [reply]
ありがとうございます！

**Comment by staticWagomU** [comment]
始めまして。
よろしければレビュアーとして参加させてください。

**Reply by azu** [reply]
ありがとうございます！

**Reply by staticWagomU** [reply]
地下鉄に乗ることが多く、web版が見れないことが多々あるため、お手数をおかけしますが、PDF版を頂いてもよろしいでしょうか？

**Reply by azu** [reply]
@staticWagomU ~~メールアドレス教えてもらっても大丈夫ですか?~~

`js-primer` orgの方に招待したので、このPrivate Disussionを参照してください。
https://github.com/orgs/js-primer/teams/reviewer/discussions/1

**Comment by kakts** [comment]
メールでも送らせていただきましたが、レビュアーとして参加させていただきたいです。
よろしくお願いいたします

**Reply by azu** [reply]
ありがとうございます！

(`info@jsprimer.net` の方にちょっとメールまだ来てないですね。転送設定がおかしいのかな…)

**Reply by kakts** [reply]
こちら参加する旨をメールしただけで、ここでコメントしたのでメール読めなくても大丈夫でした

**Comment by 2nofa11** [comment]
微力ながらレビュアーとして参加させていただければと思います。

**Reply by azu** [reply]
ありがとうございます！

**Comment by matobaa** [comment]
はじめまして。レビュー参加させていただきたいです。よろしくお願いします。

**Reply by azu** [reply]
ありがとうございます！

**Comment by r-shirasu** [comment]
初めまして！

是非参加させていただきたいです。
よろしくお願いいたします。

**Reply by azu** [reply]
ありがとうございます！

**Comment by haruguchi-yuma** [comment]
初めまして！
JS Primerには大変お世話になったので是非参加したいです。よろしくお願いします。

**Reply by azu** [reply]
ありがとうございます！

**Comment by kobakazu0429** [comment]
初めまして！
是非参加させてください！

**Reply by azu** [reply]
ありがとうございます！

**Comment by morinokami** [comment]
はじめまして！

レビュアーを募集されているとのことで、ぜひ自分も参加したいと思いコメントさせていただきました。書籍のレビューは初めてのため、どれほど力になれるかはわかりませんが、よろしくお願いします🙏

**Reply by azu** [reply]
ありがとうございます！

**Comment by azu** [comment]
一旦 10人ほど集まったので、募集はここで一区切りにしたいと思います。

レビューに参加してくれる方々ありがとうございます！

## レビューの仕方

読んでいて気になったことをコメント書いたり、明らかな間違いだったりしたらIssueやPRを出してもらう形で問題ありません。

- [ウェブ版](https://jsprimer.net/)(希望する人にはPDFをお渡しする)を見て、気になったことをコメント/Issue/Pull Requestしてもらう
  - Issueにするだけでも問題ありません
- レビュー用のDiscussionがあるので、迷ったら [第二版のレビュー #1547](https://github.com/asciidwango/js-primer/discussions/1547) に書く

## レビュアーへのお願い 🙏 

[謝辞](https://jsprimer.net/intro/#thanks)に掲載する名前またはハンドルネームをこのスレッドに返信をお願いします。
後でまとめて書籍の方へと反映したいと思います。

表記は箇条書きで掲載する形で、名前でもハンドルネームでもアカウント名でも問題ありません。

```
- <name>
```

- Issue: #1590

**Reply by staticWagomU** [reply]
- staticWagomU(林　永遠)

でお願いいたします

**Reply by 2nofa11** [reply]
* 2nofa11(ツノ)  

でお願いいたします。

**Reply by KeisukeKudo** [reply]
- keisuke kudo

でお願いします。

**Reply by haruguchi-yuma** [reply]
- haruguchi(池奥 悠馬)

でお願いします。

**Reply by kakts** [reply]
- kakts (阿久津 恵太)
でお願いいたします。

**Reply by morinokami** [reply]
- 藤野慎也 (morinokami)

でお願いします！

**Reply by azu** [reply]
@nikukyugamer @matobaa @r-shirasu @kobakazu0429 こちらの方お願いできますか?

2月いっぱいで変更をフリーズする予定なので、それまでに入れたいです。

- #1590

**Reply by r-shirasu** [reply]
遅くなり、申し訳ございません。
- r-shirasu

でお願いします。

**Reply by KeisukeKudo** [reply]
- keisuke kudo（工藤佳祐）

すみません､[こちら](https://github.com/asciidwango/js-primer/discussions/1546#discussioncomment-4946715)は無しで
こっちでお願いしたいです

**Comment by azu** [comment]
@matobaa @kakts @KeisukeKudo @staticWagomU @nikukyugamer @2nofa11 @r-shirasu @haruguchi-yuma @kobakazu0429 @morinokami それぞれレビューアーの皆さんを `js-primer` organizationの方に招待しました。
(inviteメールが飛んでいると思います。抜けていたらお知らせください。)

ほとんどのケースはPublicに書いて問題ないので、次のDiscussionやこのリポジトリのIssueなどを利用してください。

- [第二版のレビュー · Discussion #1547 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/discussions/1547)

一部PDFのやり取りとかをPrivateでするために、[`js-primer` organization](https://github.com/js-primer)に招待しています。
PDFについては、次のDiscussionを参照してください。

- https://github.com/orgs/js-primer/teams/reviewer/discussions/1
